packages/git-extractor/src/routes/github/api.ts

```diff
@@ -971,3 +975,26 @@ export async function downloadZip(
 
   return loadedZip;
 }
+
+export async function checkRemainingRateLimit(
+  numberOfRequests: number,
+): Promise<boolean> {
+  const url = "https://api.github.com/rate_limit";
+  const response: { data: { resources: { core: { remaining: number } } } } = await requestAxios(
```

Comment:
**thought:** This is my first time looking at this repo, so I don't know what lead to decisions being made. That said, if we were using [`@octokit/rest`](https://github.com/octokit/rest.js/) for making requests, we would be able to do away with per-request boilerplate, and have fully typed responses without having us write the response types. 😊

If it was a conscious decision not to use Octokit for making requests, we could definitely use their type definitions. Perhaps something worth doing down the line.

Reply:
Yah, I don't know why we didn't go with `@octokit/rest`, but there was no reason not to go with it. Since we're rewriting this service to Elixir now I'm not sure if it's still something that we want to do, but it's a good point for refactoring.